### DISCUSSION TITLE
Get data from 3rd party API into Strapi

### DISCUSSION THREAD

**Discussion by Jesus82** [strapi/strapi#6076] [Legacy Topics]
I would like to populate some of my fields with data fetched from a 3rd party API. So f.ex. I would have a field called `api_id`, and that `id` would be part of the call. Then, with the response, I would populate several fields. Finally, I would update this data periodically (f.ex, every week)

If anyone can point me a direction where to start, I would be grateful.

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by lauriejim** [comment]
Hello! This video will help you to achieve this kind of thing.

https://www.youtube.com/watch?v=4C9FgMVAAKA&list=PL7Q0DQYATmvhlHxHqfKHsr-zFls2mIVTi&index=3&t=0s

**Reply by Jesus82** [reply]
Hi, thanks for the reply. 

My question is
If I follow this method, will I store the info from the API to my Database once when it's retrieved, or does it make a call to Shopify each time there is a petition? 

For example, if there are 10 customers checking a product page, does it make an API call to Shopify each time a customer sees the page? I would rather be interested in making just one call and storing that info in my DB—and then update it in a timely basis, like for example once a week).

**Reply by derrickmehaffy** [reply]
for timely based updates you would look into a cron, however if you scale at all (PM2 clusters, or multi-server clusters) the built in cron will not work (as you would be triggering the same update on multiple servers at the same time, and thus a race condition).

For clusters you would need to handle this outside of Strapi.

Regarding the initial load, you could check to see if the data exists, if it doesn't grab it, store it, and display it. If it does exist, then simply display it.

**Reply by Jesus82** [reply]
Thanks for the comment!

In my case I need to do an http call with an id + some tokens. What would be the correct way to do it. Should I install a library like Axios, or is there any other method already available?

**Comment by lindesvard** [comment]
Is there some way of doing similar thing but for the edit view. I want to connect an article in strapi to an external api. But I do not want to look up the id beforehand and then add it manually to a uuid field. I want a field similar as relations but the data is fetched from the remote system. 

Is this possible today or do I need to wait for custom fields to be able to do this?
